sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue

```diff
@@ -85,26 +80,68 @@ export default defineComponent({
       type: String as PropType<ApplicationStatus>,
       required: true,
     },
-    statusUpdatedOn: {
-      type: String,
-      required: true,
-    },
   },
   setup(props) {
     const hasDeclinedCard = ref(false);
-    const applicationStatusTracker = ref<Record<number, ApplicationStatus>>({
+    const applicationTrackerLabels = ref<Record<number, ApplicationStatus>>({
       0: ApplicationStatus.submitted,
       1: ApplicationStatus.inProgress,
       2: ApplicationStatus.assessment,
       3: ApplicationStatus.enrollment,
       4: ApplicationStatus.completed,
     });
-    // trackFillColor will vary when more status are added.
-    const trackFillColor = ref("warning");
+    const applicationProgressDetails = ref(
+      {} as ApplicationProgressDetailsAPIOutDTO,
+    );
+    const applicationEndStatus = ref({} as ApplicationEndStatusIconDetails);
+    const trackFillColor = computed(() => {
+      if (props.applicationStatus === ApplicationStatus.completed) {
+        return "success";
+      }
+      if (applicationEndStatus.value.endStatusType === "error") {
+        return "error";
+      }
+      return "warning";
+    });
+
+    onMounted(async () => {
+      applicationProgressDetails.value =
+        await ApplicationService.shared.getApplicationProgressDetails(
+          props.applicationId,
+        );
+      // Application is complete.
+      if (
+        applicationProgressDetails.value.firstCOEStatus ===
+          COEStatus.completed &&
+        (!applicationProgressDetails.value.secondCOEStatus ||
+          applicationProgressDetails.value.secondCOEStatus ===
+            COEStatus.completed)
+      ) {
+        applicationEndStatus.value = {
+          endStatusType: "success",
+          endStatusIcon: "fa:fas fa-check-circle",
+        };
+      }
+      // One of the requests or confirmation is declined.
+      else if (
```

Comment:
IMO we can get rid of the first `if` and the second `if` can be the first one where the `else` would be the 
```t
applicationEndStatus.value = {
      endStatusType: "success",
      endStatusIcon: "fa:fas fa-check-circle",
   };
```

Reply:
By using the second if we are trying to ecape the default path where all the requests for an application are in Required Status where Icon and Icon color doesn't needs to be set.